src/box/lua/merger.c

```diff
@@ -596,10 +596,10 @@ luaL_merge_source_buffer_next(struct merge_source *base,
 		return -1;
 	}
 	--source->remaining_tuple_count;
-	source->buf->rpos = (char *) tuple_end;
 	if (format == NULL)
 		format = tuple_format_runtime;
 	struct tuple *tuple = tuple_new(format, tuple_beg, tuple_end);
+	ibuf_consume_before(source->buf, (char *)tuple_end);
```

Comment:
The cast isn't needed (I changed the argument type to `const void *`). Please update the submodule and remove the cast.

Reply:
Done.

```diff
diff --git a/src/box/lua/merger.c b/src/box/lua/merger.c
index c17671c2f..99c5b31b1 100644
--- a/src/box/lua/merger.c
+++ b/src/box/lua/merger.c
@@ -599,7 +599,7 @@ luaL_merge_source_buffer_next(struct merge_source *base,
        if (format == NULL)
                format = tuple_format_runtime;
        struct tuple *tuple = tuple_new(format, tuple_beg, tuple_end);
-       ibuf_consume_before(source->buf, (char *)tuple_end);
+       ibuf_consume_before(source->buf, tuple_end);
        if (tuple == NULL)
                return -1;
```